arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    }
+  },
+  "definitions": {
+    "Object": {
+      "type": "object",
+      "properties": {}
+    },
+    "SharedAccessSignatureAuthorizationRule": {
+      "description": "Contains properties that describe the keys to access the IotHub artifacts.",
+      "type": "object",
+      "properties": {
+        "keyName": {
+          "description": "Gets or sets the name of the key.",
+          "type": "string"
+        },
+        "primaryKey": {
+          "description": "Gets or sets the primary key.",
+          "type": "string"
+        },
+        "secondaryKey": {
+          "description": "Gets or sets the secondary key.",
+          "type": "string"
+        },
+        "rights": {
+          "description": "Gets or sets the access rights.",
+          "enum": ["RegistryRead", "RegistryWrite", "ServiceConnect", "DeviceConnect"],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "AccessRights",
+            "modelAsString": "True"
+          }
+        }
+      }
+    },
+    "IotHubProperties": {
+      "description": "Contains user facing view of Iot Hub properties",
+      "type": "object",
+      "properties": {
+        "authorizationPolicies": {
+          "description": "Gets or sets the authorization rules.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedAccessSignatureAuthorizationRule"
+          }
+        },
+        "hostName": {
+          "description": "Gets the name of the host.",
+          "type": "string"
+        },
+        "eventHubEndpoints": {
+          "description": "Gets the event hub endpoints.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/EventHubProperties"
+          }
+        },
+        "storageEndpoints": {
+          "description": "List of storage end points where files can be uploaded. Currently only one storage account can be configured.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/StorageEndpointProperties"
+          }
+        },
+        "messagingEndpoints": {
+          "description": "List of messaging end points configured.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/MessagingEndpointProperties"
+          }
+        },
+        "enableFileUploadNotifications": {
+          "description": "File upload notification is optional at iot hub level. When enabled upload notifications will be available.",
+          "type": "boolean"
+        },
+        "cloudToDevice": {
+          "$ref": "#/definitions/CloudToDeviceProperties",
+          "description": "Gets the cloud to device properties."
```

Comment:
In this case:

``` json
 "cloudToDevice": {
           "$ref": "#/definitions/CloudToDeviceProperties",
           "description": "Gets the cloud to device properties."
```
You don't want the description here (sibling properties to `$ref ` are undefined according to the swagger spec... which means we'll likely treat that as an error in a future version.